tfjs-core/src/ops/browser.ts

```diff
@@ -372,10 +373,13 @@ export async function toPixels(
 
   if (canvas != null) {
     if (!hasToPixelsWarned) {
-      console.warn(
-          'tf.browser.toPixels is not efficient to draw tensor on canvas. ' +
-          'Please try tf.browser.draw instead.');
-      hasToPixelsWarned = true;
+      const backendName = getBackend();
+      if (backendName !== 'webgl' && backendName !== 'webgpu') {
```

Comment:
How about checking like below? Then we don't need to revisit it anymore.
``` 
 const kernel = getKernel(Draw, ENGINE.backendName);
  if (kernel != null) {
        console.warn(
            'tf.browser.toPixels is not efficient to draw tensor on canvas. ' +
            'Please try tf.browser.draw instead.');
  }

Reply:
Nice catch, thanks!